src/App.tsx

```diff
@@ -15,48 +16,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SORT_FIELD_LENGTH = 'length';
+const SORT_FIELD_ALPH = 'alphabetically';
+
+interface SortParams {
+  sortField: string;
+  isReversed: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReversed }: SortParams,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((a, b) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPH:
+          return a.length - b.length;
+
+        case SORT_FIELD_LENGTH:
+          return a.localeCompare(b);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return isReversed ? preparedGoods.reverse() : preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const sortedGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
+
+  const shouldReset = isReversed || sortField;
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SORT_FIELD_LENGTH,
+          })}
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SORT_FIELD_ALPH,
+          })}
+          onClick={() => setSortField(SORT_FIELD_ALPH)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button', 'is-info', { 'is-light': !isReversed })}
+          onClick={() => setIsReversed(!isReversed)}
```

Comment:
move to `handleClickReverse` handler function 
and use like `setIsReversed(prev => !prev)`
```suggestion
          onClick={handleClickReverse}
```